R/env.R

```diff
@@ -1,191 +1,125 @@
-#' Get or create an environment.
-#'
-#' Environments are objects that create a scope for evaluation of R
-#' code. Reification of scope is one of the most powerful feature of
-#' the R language: it allows you to change what objects a function or
-#' expression sees when it is evaluated. In R, scope is hierarchical:
-#' each environment is defined with a parent environment. An
-#' environment and its grandparents form together a linear
-#' hierarchy. All objects within the grandparents are in scope unless
-#' they are eclipsed by synonyms (other bindings with the same names)
-#' in child environments.
-#'
-#' `get_env()` dispatches internally. Methods are provided for
-#' functions, formulas and frames. If called with a missing argument,
-#' the environment of the current evaluation frame (see
-#' [ctxt_stack()]) is returned. If you call `get_env()` with an
-#' environment, it acts as the identity function and the environment
-#' is simply returned (this helps simplifying code when writing
-#' generic functions).
-#'
-#' `env()` and `child_env()` create new environments. While `env()`
-#' always creates a child of the current environment, `child_env()`
-#' lets you choose a parent environment. `env_parent()` returns the
-#' parent environment of `env` if called with `n = 1`, the grandparent
-#' with `n = 2`, etc. `env_tail()` searches through the parents and
-#' returns the one which has [empty_env()] as parent.
-#'
-#' @param env An environment or an object bundling an environment:
-#'   either a [quosure] or a [closure][is_closure]. If
-#'   missing, the environment of the current evaluation frame is
-#'   returned.
-#' @param parent A parent environment. Can be an object with a S3
-#'   method for `as_env()`.
-#' @param default The default environment in case `env` does not wrap
-#'   an environment. If `NULL` and no environment could be extracted,
-#'   an error is issued.
-#' @param ...,data Uniquely named bindings. See [is_dictionaryish()].
-#'   Dots have [explicit-splicing semantics][dots_list].
-#' @param n The number of generations to go through.
-#' @seealso `scoped_env`, [env_has()], [env_assign()].
+#' Create a new environment
+#'
+#' `env()` and `child_env()` create new environments. `env()` always
+#' creates a child of the current environment while `child_env()` lets
+#' you specify a parent (see section on inheritance).
+#'
+#' @section Environments as objects:
+#'
+#' Environments are containers of uniquely named objects. Their most
+#' common use is to provide a scope for the evaluation of R
```

Comment:
I think you should say "first-class environments" somewhere in here. I think it's a good name for this property of R.